File: src/main/java/org/edtp/universe/performance/UniversePerformanceSnapshot.java
package org.edtp.universe.performance;

import org.edtp.universe.model.UniverseDimension;

import java.util.Map;
import java.util.UUID;

/** A completed one-second performance window for one universe. */
public record UniversePerformanceSnapshot(
    UUID owner,
    double budgetMillisPerSecond,
    double consumedMillisLastSecond,
    double creationMillisLastSecond,
    double averageTickMillis,
    double maximumTickMillis,
    double effectiveTps,
    long executedTicks,
    long skippedTicks,
    Map<UniverseDimension, DimensionPerformanceSnapshot> dimensions
) {
}

File: src/main/kotlin/org/edtp/universe/performance/UniversePerformanceSnapshot.kt
package org.edtp.universe.performance

import org.edtp.universe.model.UniverseDimension
import java.util.UUID

data class UniversePerformanceSnapshot(
    val owner: UUID,
    val budgetMillisPerSecond: Double,
    val consumedMillisLastSecond: Double,
    val averageTickMillis: Double,
    val maximumTickMillis: Double,
    val effectiveTps: Double,
    val executedTicks: Long,
    val skippedTicks: Long,
    val dimensions: Map<UniverseDimension, DimensionPerformanceSnapshot>,
)

data class DimensionPerformanceSnapshot(
    val averageTickMillis: Double,
    val maximumTickMillis: Double,
    val executedTicks: Long,
    val skippedTicks: Long,
)
